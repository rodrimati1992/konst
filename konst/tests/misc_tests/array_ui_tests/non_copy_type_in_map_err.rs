struct Foo<T>(T);

const _: [Foo<u32>; 3] = konst::array::map!([3, 5, 8], Foo);

const _: [Foo<String>; 1] = konst::array::map!([String::new()], Foo);

fn main(){}

