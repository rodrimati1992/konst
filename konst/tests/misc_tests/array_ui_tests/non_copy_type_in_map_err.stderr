error[E0277]: the trait bound `Foo<{integer}>: Copy` is not satisfied
 --> tests/misc_tests/array_ui_tests/non_copy_type_in_map_err.rs:3:26
  |
3 | const _: [Foo<u32>; 3] = konst::array::map!([3, 5, 8], Foo);
  |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Copy` is not implemented for `Foo<{integer}>`
  |
note: required by a bound in `konst_kernel::macros::array_macros::uninit_copy_array_of_len`
 --> $WORKSPACE/konst_kernel/src/macros/array_macros.rs
  |
  |     U: Copy,
  |        ^^^^ required by this bound in `konst_kernel::macros::array_macros::uninit_copy_array_of_len`
  = note: this error originates in the macro `$crate::__array_map` which comes from the expansion of the macro `konst::array::map` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider annotating `Foo<{integer}>` with `#[derive(Copy)]`
   |
1  | #[derive(Copy)]
   |

error[E0277]: the trait bound `Foo<String>: Copy` is not satisfied
 --> tests/misc_tests/array_ui_tests/non_copy_type_in_map_err.rs:5:29
  |
5 | const _: [Foo<String>; 1] = konst::array::map!([String::new()], Foo);
  |                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Copy` is not implemented for `Foo<String>`
  |
note: required by a bound in `konst_kernel::macros::array_macros::uninit_copy_array_of_len`
 --> $WORKSPACE/konst_kernel/src/macros/array_macros.rs
  |
  |     U: Copy,
  |        ^^^^ required by this bound in `konst_kernel::macros::array_macros::uninit_copy_array_of_len`
  = note: this error originates in the macro `$crate::__array_map` which comes from the expansion of the macro `konst::array::map` (in Nightly builds, run with -Z macro-backtrace for more info)
